server/plugin.go

```diff
@@ -425,3 +426,65 @@ func (p *Plugin) CheckSiteURL() error {
 	}
 	return nil
 }
+
+func (p *Plugin) GetWebhookURL(jiraURL string, teamID, channelID string) (subURL, legacyURL string, err error) {
+	cf := p.getConfig()
+
+	instanceID, err := p.ResolveWebhookInstanceURL(jiraURL)
+	if err != nil {
+		return "", "", err
+	}
+
+	team, appErr := p.API.GetTeam(teamID)
+	if appErr != nil {
+		return "", "", appErr
+	}
+
+	channel, appErr := p.API.GetChannel(channelID)
+	if appErr != nil {
+		return "", "", appErr
+	}
+
+	v := url.Values{}
+	secret, _ := url.QueryUnescape(cf.Secret)
+	v.Add("secret", secret)
+	subURL = p.GetPluginURL() + instancePath(routeAPISubscribeWebhook, instanceID) + "?" + v.Encode()
+
+	// For the legacy URL, add team and channel. Secret is already in the map.
+	v.Add("team", team.Name)
+	v.Add("channel", channel.Name)
+	legacyURL = p.GetPluginURL() + instancePath(routeIncomingWebhook, instanceID) + "?" + v.Encode()
+
+	return subURL, legacyURL, nil
+}
+func newWebhook(jwh *JiraWebhook, eventType string, format string, args ...interface{}) *webhook {
+	return &webhook{
+		JiraWebhook: jwh,
+		eventTypes:  NewStringSet(eventType),
+		headline:    jwh.mdUser() + " " + fmt.Sprintf(format, args...) + " " + jwh.mdKeySummaryLink(),
+	}
+}
+
+func (p *Plugin) getConnection(instance Instance, notification webhookUserNotification) (con *Connection, err error) {
+	var mattermostUserID types.ID
+
+	// prefer accountId to username when looking up UserIds
+
+	if notification.jiraAccountID != "" {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraAccountID)
+	} else {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraUsername)
+	}
+	if err != nil {
+		return
+	}
+
+	// Check if the user has permissions.
+	con, err = p.userStore.LoadConnection(instance.GetID(), mattermostUserID)
+	if err != nil {
+		// Not connected to Jira, so can't check permissions
+		return
```

Comment:
seems logging and returning error can help to understand better the situation

Reply:
done i was add the logging